controllers/spifilecontentrequest_controller.go

```diff
@@ -188,10 +191,16 @@ func (r *SPIFileContentRequestReconciler) Reconcile(ctx context.Context, req ctr
 		// check if the URL is processable and create binding, otherwise fail fast
 		sp, err := r.ServiceProviderFactory.FromRepoUrl(ctx, request.Spec.RepoUrl, request.Namespace)
 		if err != nil {
-			lg.Error(err, "unable to get the service provider for the SPIFileContentRequest")
-			// we determine the service provider from the URL in the spec. If we can't do that, nothing works until the
-			// user fixes that URL. So no need to repeat the reconciliation and therefore no error returned here.
-			r.updateFileRequestStatusError(ctx, &request, noSuitableServiceProviderFound)
+			var validationErr validator.ValidationErrors
+			if stderrors.As(err, &validationErr) {
+				lg.Error(err, "unable to validate the service provider for the SPIFileContentRequest")
```

Comment:
it should unwrap the  error., so we'll see the details. I don't want to be so specific there because potentially other kind of validations may be done as well